src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,424 @@
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.config.commands.CommandCategory
+import at.hannibal2.skyhanni.config.commands.CommandRegistrationEvent
+import at.hannibal2.skyhanni.config.features.inventory.CakeTrackerConfig.CakeTrackerDisplayOrderType
+import at.hannibal2.skyhanni.config.features.inventory.CakeTrackerConfig.CakeTrackerDisplayType
+import at.hannibal2.skyhanni.config.storage.ProfileSpecificStorage.CakeData
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.ConfigLoadEvent
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.features.inventory.patternGroup
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.ConditionalUtils
+import at.hannibal2.skyhanni.utils.HypixelCommands
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.InventoryUtils.getAllItems
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.formatInt
+import at.hannibal2.skyhanni.utils.RegexUtils.matchGroup
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockTime
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import net.minecraft.inventory.ContainerChest
+import net.minecraft.inventory.Slot
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.time.Duration.Companion.milliseconds
+
+private typealias DisplayOrder = CakeTrackerDisplayOrderType
+private typealias DisplayType = CakeTrackerDisplayType
+
+@SkyHanniModule
+object CakeTracker {
+
+    private val storage get() = ProfileStorageData.profileSpecific?.cakeData
+    private val config get() = SkyHanniMod.feature.inventory.cakeTracker
+
+    private var currentYear = 0
+    private var inCakeBag = false
+    private var inCakeInventory = false
+    private var timeOpenedCakeInventory = SimpleTimeMark.farPast()
+    private var inAuctionHouse = false
+    private var auctionCakesCache = mapOf<Int, Boolean>()
+    private var searchingForCakes = false
+    private var knownCakesInCurrentInventory = listOf<Int>()
+
+    private var cakeRenderables = listOf<Renderable>()
+    private var lastKnownCakeDataHash = 0
+
+    private var unobtainedHighlightColor: Color = config.unobtainedAuctionHighlightColor.get().toChromaColor()
```

Comment:
if done this way, chroma colors wont work. instead of having them be properties and store them as vars, just replace these with getters, like
```kt
    private val unobtainedHighlightColor: Color get() = config.unobtainedAuctionHighlightColor.toChromaColor()
```
both the unobtained and obtained colors need this change